design-toolkit/components/src/components/menu/index.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import ChevronRight from '@accelint/icons/chevron-right';
+import { createContext, useCallback, useContext } from 'react';
+import {
+  Header as AriaHeader,
+  Menu as AriaMenu,
+  Collection as AriaMenuCollection,
+  MenuItem as AriaMenuItem,
+  type MenuItemRenderProps as AriaMenuItemRenderProps,
+  MenuSection as AriaMenuSection,
+  MenuTrigger as AriaMenuTrigger,
+  type MenuTriggerProps as AriaMenuTriggerProps,
+  Separator as AriaSeparator,
+  SubmenuTrigger as AriaSubmenuTrigger,
+  Text as AriaText,
+  type ContextValue,
+  Popover,
+} from 'react-aria-components';
+import { callRenderProps, isSlottedContextValue } from '../../lib/utils';
+import { Icon } from '../icon';
+import { MenuStyles, MenuStylesDefaults } from './styles';
+import type {
+  MenuIconProps,
+  MenuItemProps,
+  MenuProps,
+  MenuSectionProps,
+  MenuTextProps,
+  SeparatorProps,
+} from './types';
+
+const { item, label, description, menu, more, icon, separator } = MenuStyles();
+
+export const MenuContext =
+  createContext<ContextValue<MenuProps<unknown>, HTMLDivElement>>(null);
+
+export function Menu<T extends object>(props: MenuProps<T>) {
```

Comment:
The menu should be using the menu context as well in case a parent is dictating props